src/lib/proof-system/zkprogram.ts

```diff
@@ -229,40 +258,13 @@ let SideloadedTag = {
  * @returns an object that can be used to compile, prove, and verify the program.
  */
 function ZkProgram<
-  Config extends {
-    publicInput?: ProvableType;
-    publicOutput?: ProvableType;
-    methods: {
-      [I in string]: {
-        privateInputs: Tuple<PrivateInput>;
-        auxiliaryOutput?: ProvableType;
-      };
-    };
-  },
-  Methods extends {
-    [I in keyof Config['methods']]: Method<
-      InferProvableOrUndefined<Get<Config, 'publicInput'>>,
-      InferProvableOrVoid<Get<Config, 'publicOutput'>>,
-      Config['methods'][I]
-    >;
-  },
-  // derived types for convenience
-  MethodSignatures extends Config['methods'] = Config['methods'],
-  PrivateInputs extends {
-    [I in keyof Config['methods']]: Config['methods'][I]['privateInputs'];
-  } = {
-    [I in keyof Config['methods']]: Config['methods'][I]['privateInputs'];
-  },
-  AuxiliaryOutputs extends {
-    [I in keyof MethodSignatures]: Get<MethodSignatures[I], 'auxiliaryOutput'>;
-  } = {
-    [I in keyof MethodSignatures]: Get<MethodSignatures[I], 'auxiliaryOutput'>;
-  }
+  Config extends ConfigBaseType,
+  _ extends unknown = unknown // weird hack that makes methods infer correctly when their inputs are not annotated
```

Comment:
this extra line fixed the type error in `foreign-field-gadgets.unit-test.ts`

I also noticed that you could remove the errors when explicitly annotating method inputs in that file. Which would be good practice, since if you don't do it the inputs will be typed as `any` inside the method.

however, I also found that adding _any_ extra generic type argument makes the error disappear. no idea why!
since writing a zkprogram is already hard, because you tend to get confusing type errors on an incomplete ZkProgram definition, I think it's worth keeping this hack to remove one more confusing error.